packages/auth/src/grant/service.ts

```diff
@@ -24,14 +25,15 @@ export interface GrantService {
     interactId: string,
     interactNonce: string,
     options?: GetByInteractionSessionOpts
-  ): Promise<PendingGrant | undefined>
-  issueGrant(grantId: string): Promise<Grant>
+  ): Promise<InteractiveGrant | undefined>
+  markPending(grantId: string): Promise<Grant>
```

Comment:
What would be a reason to manually mark a grant as pending?

Reply:
I thought it would make sense to mark a grant as pending only when an interaction is initiated, so it would have been called on the `startInteraction` route. Looking at the GNAP spec more closely though, it implies that a Grant should be marked pending as soon as it knows whether or not interaction is required, irrespective of if it has been initiated or not. I'll remove this in light of that and have the Grant be marked `Pending` on creation if it needs interaction.